src/hex.py

```diff
@@ -10,13 +10,20 @@
 
 command, x = sys.argv[1:3]
 
+
 match command:
     case "encode":
-        # Implement the encoding here
-        encoding = ""
+    y=[]
+    for i in range(len(x)):
+        n=hex(ord(x[i]))
+        y.append(n)
+        encoding = " ".join(y)
         print(encoding)
```

Comment:
Are you printing for each iteration of the loop? Shouldn't you wait until the loop was done?